libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -449,6 +455,35 @@ constructor(
         tripSession.unregisterStateObserver(tripSessionStateObserver)
     }
 
+    /**
+     * Attach your own controller to determine when drivers arrived at stops via [ArrivalController]
+     * Use [navigateToNextStop] to manually move navigator to the next stop.
+     *
+     * @param arrivalController ArrivalObserver
+     */
+    fun attachArrivalController(arrivalController: ArrivalController) {
+        arrivalProgressObserver.attach(arrivalController)
+    }
+
+    /**
+     * Stop controlling arrival at stops, detaching will require the caller
+     * to call [navigateToNextStop] manually.
+     */
+    fun removeArrivalController() {
+        tripSession.unregisterRouteProgressObserver(arrivalProgressObserver)
+        arrivalProgressObserver.detach()
+    }
+
+    /**
+     * After arriving a stop, this can be used to manually decide when to start
+     * navigating to the next stop. Use the [ArrivalController] to require manual navigation.
```

Comment:
`manual` as in "automatic" increment of the leg index?

Reply:
hmm neither? mostly wasnt to reference the ArrivalController from the comment